lib/iris/tests/unit/util/test_flatten_multidim_coord.py

```diff
@@ -0,0 +1,96 @@
+# (C) British Crown Copyright 2010 - 2018, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Test function :func:`iris.util.flatten_multidim_coord`."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import unittest
+
+import iris.tests.stock as stock
+from iris.coords import DimCoord, AuxCoord
+from iris.util import flatten_multidim_coord, flatten_cube
+
+
+class Test(tests.IrisTest):
+    def test_argument_is_basestring(self):
+        cube_a = stock.simple_2d_w_multidim_coords()
+
+        cube_b = flatten_multidim_coord(cube_a, 'bar')
+        self.assertEqual(cube_b.shape, (12, ))
+
+    def test_argument_is_coord_instance(self):
+        cube_a = stock.simple_3d_w_multidim_coords()
+        coord = cube_a.coord('bar').copy()
+        cube_b = flatten_multidim_coord(cube_a, coord)
+        self.assertEqual(cube_b.dim_coords, (cube_a.coord('wibble'), ))
+        self.assertEqual(cube_b.shape, (2, 12))
```

Comment:
I also think that these tests are getting a bit jumbled.  Your top two tests here are `test_argument_is_basestring` and `test_argument_is_coord_instance`, which should have all the same parameters and checks (i.e. the same stock cube and the same asserts) to make sure that the behaviour is exactly the same when that one input type is changed.

Then it is appropriate to add another test with a 3d cube to make sure that this is also handled correctly, rather than adding those checks sneakily into a test for something else. 

Would you mind separating out these tests in that way please?